pkg/services/ngalert/schedule/recording_rule.go

```diff
@@ -205,17 +206,19 @@ func (r *recordingRule) tryEvaluation(ctx context.Context, ev *Evaluation, logge
 		return fmt.Errorf("the query failed with an error: %w", err)
 	}
 
-	logger.Info("Recording rule evaluated", "results", result, "duration", evalDur)
+	logger.Debug("Recording rule query completed", "resultCount", len(result.Responses), "duration", evalDur)
 	span := trace.SpanFromContext(ctx)
-	span.AddEvent("rule evaluated", trace.WithAttributes(
+	span.AddEvent("query succeeded", trace.WithAttributes(
 		attribute.Int64("results", int64(len(result.Responses))),
 	))
 
 	frames, err := r.frameRef(ev.rule.Record.From, result)
 	if err != nil {
-		span.SetStatus(codes.Error, "failed to extract frames from rule evaluation")
-		span.RecordError(err)
-		return fmt.Errorf("failed to extract frames from rule evaluation: %w", err)
+		span.AddEvent("query returned no data, nothing to write", trace.WithAttributes(
```

Comment:
An error in this case seems to be more than just Nodata. Perhaps use some generic message?

Reply:
The only reason i included it is because the error string tells what specifically about the frame we didn't like - specifically it tells us if a refID is missing from the query data response entirely vs present with empty frames. It's subtle, but i thought that might be nice for debugging.

Better suited for debug logs instead of trace perhaps?